core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -634,6 +634,16 @@ class MapUnit {
         attacksThisTurn = 0
         due = true
 
+        // Hakkapeliitta movement boost
+        if(hasUnique("Bonus for units in 2 tile radius 15%")) {
```

Comment:
I would use `getMatchingUniques("Great Person - []").any { it.params[0] == "War" }` here, as that is the more general form of great generals.

Reply:
ok, done